cpp/examples/set_cuda_architecture.cmake

```diff
@@ -0,0 +1,37 @@
+# =============================================================================
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+# get the CUDF_TAG from fetch_dependencies.cmake (remove all whitespace & parse line: set(CUDF_TAG
+# ..))
+execute_process(
+  COMMAND
+    bash -c
+    "sed '/^$/d;s/[[:blank:]]//g' ./fetch_dependencies.cmake | grep 'set(CUDF_TAG' | sed 's/.*CUDF_TAG//' | sed 's/.$//'"
```

Comment:
Instead of `grep`ping and `sed`ing a CMake file, any data should be stored in either a `.cmake` file that sets variables or a JSON file that can be read with `string(JSON)`. Then `fetch_dependencies.cmake` and `set_cuda_architecture.cmake` can both read from this single source of truth.

Reply:
The reason to keep `CUDF_TAG` in `fetch_dependencies.cmake was to not break any other CI artifacts reading from here like: https://github.com/rapidsai/cudf/blob/79d2dba141e0cbc464810a0b214d2c714ebe1a18/ci/release/update-version.sh#L72-L73. Let me just avoid all this everywhere.

Reply:
All done!